freqtrade/optimize/backtesting.py

```diff
@@ -221,7 +221,21 @@ def backtest(self, args: Dict) -> DataFrame:
 
             # Convert from Pandas to list for performance reasons
             # (Looping Pandas is slow.)
-            ticker = [x for x in ticker_data.itertuples()]
+
+            #ticker = [x for x in ticker_data.itertuples()]
+            """
+            Use faster numpy to mask out rows we will not use in slower itertuples loop
+
+            Filter rows that are buy or sell to be processed
+            Also keep the last row from in any state for open trades
+            """
+            mask = (ticker_data['buy'].values != 0) | (ticker_data['sell'].values != 0)
```

Comment:
Are you sure this works with stoploss / ROI? i think you remove lines which should be passed to `_get_sell_trade_entry` ...